riscv-c-api.md

```diff
@@ -114,7 +114,221 @@ For example:
 * `__attribute__((interrupt("supervisor")))`
 * `__attribute__((interrupt("machine")))`
 
-## Intrinsic Functions
 
-Do we really have none of these?  I can't figure out
-`gcc/gcc/config/riscv/riscv-builtins.c`...
+## Scalar Cryptography Extension Builtins
+
+This is a design document, and the implementation in compilers may be partial.
+However, if architecture test macros are defined, the extensions are available.
+
+For post-ratification Scalar Cryptography v1.0, the Arch Version is `1000000`
+(one million).
+
+| Name          | Value        | When defined                                                 |
+| ------------- | ------------ | ------------------------------------------------------------ |
+| __riscv_zk    | Arch Version | `Zbkb` `Zbkc` `Zbkx` `Zkne` `Zknd` `Zknh` are all available. |
+| __riscv_zkn   | Arch Version | `Zbkb` `Zbkc` `Zbkx` `Zkne` `Zknd` `Zknh` are all available. |
+| __riscv_zks   | Arch Version | `Zbkb` `Zbkc` `Zbkx` `Zksed` `Zksh` are all available.       |
+| __riscv_zbkb  | Arch Version | `Zbkb` extension is available.                               |
+| __riscv_zbkc  | Arch Version | `Zbkc` extension is available.                               |
+| __riscv_zbkx  | Arch Version | `Zbkx` extension is available.                               |
+| __riscv_zknd  | Arch Version | `Zknd` extension is available.                               |
+| __riscv_zkne  | Arch Version | `Zkne` extension is available.                               |
+| __riscv_zknh  | Arch Version | `Zknh` extension is available.                               |
+| __riscv_zksed | Arch Version | `Zksed` extension is available.                              |
+| __riscv_zksh  | Arch Version | `Zbsh` extension is available.                               |
+
+The extensions `Zkt` or `Zkr` do not introduce builtins and do not have
+architecture extension test macros.
+
+However, **all** of the scalar cryptography builtins below must offer the
+same data-independent latency guarantees as the `Zkt` extension. The user
+must be able to trust that the compiler can't use table lookups, conditional
+branching, etc when transforming these crypto builtins into code.
+
+Shorter intrinsics can be be mapped to builtins via the `rvkintrin.h` system
```

Comment:
`rvkintrin.h` seems only mention here, but no detail info, I would we suggest define which API/intrinsic has exposed in this header instead of defining low-level built-in function.

Reply:
Added the file from https://github.com/rvkrypto/rvkrypto-fips/  that defineds a mapping. That code has been significantly refactored now.